_docs/concepts/encryption.md

```diff
@@ -0,0 +1,197 @@
+>
+> Used to Generate:
+> 
+> MAC<sub>K<sub>vol</sub></sub> = Keyed digest of volume master key  
+> 
+> Through:
+> 
+> MAC<sub>K<sub>vol</sub></sub> = HMAC<sub>IV<sub>vol</sub></sub>(K<sub>vol</sub>) 
+
+Finally, K<sub>user</sub>, MAC<sub>K<sub>vol</sub></sub>,
+K<sub>user<sub>vol</sub></sub> and IV<sub>vol</sub> are stored in a Kubernetes
+secret (see [Encrypted Volumes](/docs/operations/encrypted-volumes) for best
+practices regarding backing up StorageOS secrets).
+
+## Volume Master Key Derivation
+
+K<sub>vol</sub> is used to encrypt and decrypt volumes, it is derived using
+K<sub>user<sub>vol</sub></sub> and IV<sub>vol</sub>.
+
+>K<sub>user<sub>vol</sub></sub> = Volume User Key - Stored in Kubernetes secret
+>
+>IV<sub>vol</sub> = Volume initialisation vector - Stored in Kubernetes secret
+>
+>Used to generate:
+>
+>K\`<sub>vol</sub> = Derived Volume Master key - Used to encrypt the volume
+>
+>Through:
+>
+>K<sub>i</sub> = HKDF(K<sub>user</sub>, IV<sub>vol</sub>)
+>
+>K\`<sub>vol</sub> = D<sub>K<sub>i</sub></sub>(K<sub>user<sub>vol</sub></sub>)
+
+We verify that K<sub>vol</sub> is correct by creating a new keyed digest of the
+derived K<sub>vol</sub> using IV<sub>vol</sub> and comparing it against the digest that
+was stored when the volume was created.
+
+> K\`<sub>vol</sub> = Derived K<sub>vol</sub>
+>
+> IV<sub>vol</sub> = Initialisation vector - Stored in Kubernetes secret
+>
+> MAC<sub>K<sub>vol</sub></sub> = Volume master key digest - Stored in
+> Kubernetes secret
+>
+> Assert:
+>
+> HMAC<sub>IV<sub>vol</sub></sub>(K\`<sub>vol</sub>) === MAC<sub>K<sub>vol</sub></sub>
+
+## Writing to an encrypted volume
+
+When data is written to an encrypted volume, data is encrypted using the volume
+master key K<sub>vol</sub>.
+
+>Volume Master Key = K<sub>vol</sub> - Derived from
+>K<sub>user<sub>vol</sub></sub>
+>
+>Disk block number = I<sub>blk</sub>
+>
+>Plain text block, size len bytes = P<sub>blk</sub>
+>
+>Tweak = T<sub>blk</sub> - Derived using whatever mapping the system required (MORE NEEDED
+>HERE)
+>
+>Used to generate:
+>
+>Cipher text block = C<sub>blk</sub>
+>
+>Through:
+>
+>C<sub>blk</sub> = Exts(kvol, T<sub>blk</sub>, len)(P<sub>blk</sub>)
+
+## Reading from an encrypted volume
+
+When data is read from an encrypted volume, the data is decrypted using the
+volume master key (K<sub>vol</sub>). K<sub>vol</sub> is derived form the volume admin key (K<sub>admvol</sub>)
+and the Volume initialisation vector (IV<sub>vol</sub>)
+
+>Volume Master Key = K<sub>vol</sub> 
+>
+>Disk block number to be decrypted = I<sub>blk</sub>Ciphertext block (WHAT IS THIS USED
+>FOR?)
+>
+>Tweak = T<sub>blk</sub> - Derived using whatever mapping the system required (MORE NEEDED
+>HERE)
+>
+>Cipher text block, size len bytes = C<sub>blk</sub>
+>
+>Used to generate:
+>
+>Plain text block = P<sub>blk</sub>
+>
+>P<sub>blk</sub> = D<sub>XTS</sub>(K<sub>vol</sub>, T<sub>blk</sub>, len)(C<sub>blk</sub>)
+
+
+## Recovery Keys
```

Comment:
Avoid mentioning this and key rotation for now - if it comes up in conversation we can talk about it.